components/script/dom/url.rs

```diff
@@ -191,6 +202,11 @@ impl URLMethods for URL {
         UrlHelper::SetSearch(&mut self.url.borrow_mut(), value);
```

Comment:
Instead of calculating query pairs on-demand, I'd rather stick to the spec and store them immediately.

Regarding iteration, this is something we currently don't handle - it's commented in URLSearchParams.webidl:
`// iterable<USVString, USVString>;`
Please correct me if I'm wrong.

Firefox's approach looks reasonable to me. We'll just have to be careful when implementing the iterable interface.